src/pybind/mgr/ansible_orchestrator/ansible_runner_svc.py

```diff
@@ -0,0 +1,278 @@
+        a event filter
+
+        @returns: the events that matches with the patterns provided
+        """
+
+        if not self.result_task_pattern or not self.play_uuid:
+            result_events = {}
+
+        response = self.rest_client.http_get(PLAYBOOK_EVENTS % self.play_uuid)
+
+        if not response:
+            result_events = {}
+        else:
+            events = json.loads(response.text)["data"]["events"]
+            result_events = {event:data for event,data in events.iteritems()
+                            if "task" in data and
+                            re.match(self.result_task_pattern, data["task"])}
+            if event_filter:
+                result_events = {event:data for event,data in result_events.iteritems()
+                                if re.match(event_filter, data['event'])}
+
+        self.log.info("Requested playbook result is: %s", json.dumps(result_events))
+        return result_events
+
+class Client(object):
+    """An utility object that allows to connect with the Ansible runner service
+    and execute easily playbooks
+    """
+
+    def __init__(self, server, port, user, password, certificate, logger):
+        """Get the Ansible runner service url and user credentials to be used in
+        all the requests.
+
+        @param ar_service_url: The URL of the Ansible runner service
+        @param user: A tuple with (user_name, password), the user authorized to
+                    connect with the Ansible Runner service
+        """
+        self.server = server
+        self.port = port
+        self.user = user
+        self.password = password
+        self.log = logger
+        self.auth = (self.user, self.password)
+
+        # Used to verify or not https server identity
+        if not certificate:
+            self.certificate = False
+        else:
+            self.certificate = certificate
+
+        # Once authenticated this token will be used in all the requests
+        self.token = ""
+
+        self.server_url = "https://{0}:{1}".format(self.server, self.port)
+
+        # Log in the server and get a token
+        self.login()
+
+    def login(self):
+        """ Login with user credentials to obtain a valid token
+        """
+
+        response = None
+
+        try:
+            the_url = "%s/%s" % (self.server_url, LOGIN_URL)
+            r = requests.get(the_url,
+                             auth = self.auth,
+                             verify = self.certificate)
+
+            if r.status_code != requests.codes.ok:
+                self.log.error("login error <<%s>> (%s):%s",
+                                the_url, r.status_code, r.text)
+            else:
+                self.log.info("login succesful <<%s>> (%s):%s",
+                               the_url, r.status_code, r.text)
+
+            response = r
+
+        except Exception as ex:
+            self.log.error("Ansible runner service - Unexpected error: %s", ex)
```

Comment:
if you call `log.execption`, it will also print the stack trace. 
```python
self.log.exception("Ansible runner service")
```

Reply:
Done!